### PR TITLE
8299592: Fix and reenable warnings in java.desktop native code compilation

### PR DESCRIPTION
**Pull request by djelinski**
Please review this patch that fixes and re-enables a few warnings in libawt compilation.

Verified that debug and release builds finish successfully on Win, Mac and Linux. Also verified that client libs tests still pass.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8299592](https://bugs.openjdk.org/browse/JDK-8299592): Fix and reenable warnings in java.desktop native code compilation

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [861d6eed](https://git.openjdk.org/jdk/pull/11841/files/861d6eeda2291766115c3a74bc53bcf5fd232964)
 * @SWinxy (no known openjdk.org user name / role) ⚠️ Review applies to [57be49b1](https://git.openjdk.org/jdk/pull/11841/files/57be49b1206fb2d415f5057cab9c309f6771f229)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/11841/head:pull/11841` \
`$ git checkout pull/11841`

Update a local copy of the PR: \
`$ git checkout pull/11841` \
`$ git pull https://git.openjdk.org/jdk.git pull/11841/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11841`

View PR using the GUI difftool: \
`$ git pr show -t 11841`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11841.diff">https://git.openjdk.org/jdk/pull/11841.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/11841#issuecomment-1370926218)

### CODE REVIEW

**Review by magicus — APPROVED** [review]
Build changes look great! I have also looked through the code changes itself, and they look sane to me, but you'll need an approval from someone on the client team as well.

**Comment by djelinski** [comment]
In case it helps with the review, most of the fixes do not change the behavior, but make the compiler happy:
- the changes to SAFE_TO_ALLOC: assuming that no negative values are passed to the macros, the end result will be the same (modulo that the SAFE_TO_ALLOC macros should probably use `>=` instead of `>`)
- adding `inline` to `static`: `inline` is a C11 thing, and is generally ignored by the compiler, but `static inline` functions may be unused and plain `static` trigger a warning when unused
- removed a few unused functions, moved `DTrace_PrintStdErr` into `#ifdef DEBUG`
- replaced `double` functions with `float` where the argument was a `float` and the result was stored in a `float`
- fixed the type of `PFNISTHEMEBACKGROUNDPARTIALLYTRANSPARENT` (this was only reported as a warning when compiler was run with `-analyze` parameter) (given that we dropped support for Win7 and older, can we link to uxtheme instead of checking for function availability?)
- removed `extern` on `DMemStatePtr` definition; it is still present on the declaration in header file, and definitions should not be `extern`

The following fixes may change behavior:
- changed `DMem_DumpHeader` to use snprintf, adjusted the buffer size to use what gcc suggested (may avoid crashing on long input)
- fixed error handling in `D3DBlitTextureToSurface` (`d3dc->SetTexture` failures were ignored) (warning only reported with `-analyze`)

**Comment by djelinski** [comment]
keep alive / waiting for review from client-libs...

**Review by SWinxy — APPROVED** [review]
These changes look fine to me as well. Built and ran a couple dozen tests. `D3DBlitLoops.cpp` looks to have had a hidden bug that is now fixed with this patch. Granted, I am not to be trusted with native code.

File: src/java.desktop/share/native/common/awt/debug/debug_mem.c
Comment (aivanov-jdk) [inline src/java.desktop/share/native/common/awt/debug/debug_mem.c:72]: As far as all's aligned, should there be another space to align the type names?

File: make/modules/java.desktop/lib/Awt2dLibraries.gmk
Comment (aivanov-jdk) [inline make/modules/java.desktop/lib/Awt2dLibraries.gmk:141]: I see you declared the functions as `inline`, it was probably the intention for declaring them in the header file.

Did it not resolve `deprecated-non-prototype` warning? If not, did you submit another bug to address it?

Comment (djelinski) [reply make/modules/java.desktop/lib/Awt2dLibraries.gmk:141]: Thanks for reviewing!

I didn't see the warning in mach5 build. However, the warning was not related to inline declarations. See [JDK-8300169](https://bugs.openjdk.org/browse/JDK-8300169) where the warning was reported; the compiler complained that `sMlibFns.fptr` does not have a prototype. See the declaration:
https://github.com/openjdk/jdk/blob/0deb648985b018653ccdaf193dc13b3cf21c088a/src/java.desktop/share/native/libawt/awt/medialib/awt_ImagingLib.h#L35
I'm pretty sure that given the right compiler, the warning would still appear.

After I filed the ticket for this change, I found [JDK-8074823](https://bugs.openjdk.org/browse/JDK-8074823); let's use that to keep track of the remaining warnings.

**Review by aivanov-jdk — APPROVED** [review]
Looks good to me.

I ran builds and automated client tests, no failures detected.

**Comment by djelinski** [comment]
Thanks for the reviews!

/integrate

**Pull request closed without merging**
